src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -848,7 +849,13 @@ void GcsActorManager::ReconstructActor(const ActorID &actor_id, bool need_resche
           // if actor was an detached actor, make sure to destroy it.
           // We need to do this because detached actors are not destroyed
           // when its owners are dead because it doesn't have owners.
-          if (actor->IsDetached()) DestroyActor(actor_id);
+          if (actor->IsDetached()) {
+            DestroyActor(actor_id);
+            // NOTE: This method won't be invoked inside DestroyActor if the actor is
+            // detached because the actor state is already DEAD, so we should invoke here.
+            // TODO(sang): Refactor this to be cleaner.
+            NotifyActorDeathToListener(actor_id);
```

Comment:
```suggestion
            NotifyActorDeathListeners(actor_id);
```

Reply:
Should we be calling this at the top of the reconstruction method (unconditionally trigger listeners regardless of the cause?) It's a bit odd to only call it on final death. Presumably, the placement group should be destroyed even if the actor is reconstructable.

Reply:
Ah, yeah that's a great point. It should definitely be called before reconstruction. 

One maintainability concern is that we should keep this method idempotent (since this can be called multiple times in this scenario). I will probably change the method to be more specific to Placement group removal instead of make it a general listener (which can break idempotency easily by other developers who don't have context).

Reply:
Sounds good, so something like DestroyOwnedPlacementGroupsIfNeeded()?

Reply:
Yep will do that!